xarray/backends/scipy_.py

```diff
@@ -37,30 +37,39 @@ def __init__(self, variable_name, datastore):
 
     @property
     def array(self):
-        return self.datastore.ds.variables[self.variable_name].data
+        with self.datastore.ensure_open(autoclose=True):
+            return self.datastore.ds.variables[self.variable_name].data
 
     @property
     def dtype(self):
-        # always use native endianness
-        return np.dtype(self.array.dtype.kind + str(self.array.dtype.itemsize))
+        with self.datastore.ensure_open(autoclose=True):
```

Comment:
The `NumpyIndexingAdapter` class doesn't allow setting the `dtype` property in the constructor.  Should I inherit something else?  I'm missing some key info on design intent here.